addons/sale/models/sale_order.py

```diff
@@ -375,26 +375,26 @@ def _compute_validity_date(self):
     def _compute_journal_id(self):
         self.journal_id = False
 
-    @api.depends('partner_id')
+    @api.depends('partner_id', 'company_id')
     def _compute_note(self):
         use_invoice_terms = self.env['ir.config_parameter'].sudo().get_bool('account.use_invoice_terms')
         if not use_invoice_terms:
             return
         for order in self:
             order = order.with_company(order.company_id)
-            if order.terms_type == 'html' and self.env.company.invoice_terms_html:
+            if order.terms_type == 'html' and order.company_id.invoice_terms_html:
                 baseurl = html_keep_url(order._get_note_url() + '/terms')
                 context = {'lang': order.partner_id.lang or self.env.user.lang}
                 order.note = _('Terms & Conditions: %s', baseurl)
                 del context
-            elif not is_html_empty(self.env.company.invoice_terms):
+            elif not is_html_empty(order.company_id.invoice_terms):
                 if order.partner_id.lang:
                     order = order.with_context(lang=order.partner_id.lang)
-                order.note = order.env.company.invoice_terms
+                order.note = order.company_id.invoice_terms
 
     @api.model
     def _get_note_url(self):
-        return self.env.company.get_base_url()
+        return self.company_id.get_base_url()
```

Comment:
Doesn't make sense with the `api.model` decorator

Reply:
Indeed.
Furthermore, this change has no impact since the environment company is the order company (thanks to `order = order.with_company(order.company_id)`)